docker/labels.go

```diff
@@ -11,10 +11,11 @@ type Label string
 
 // Libcompose default labels.
 const (
-	NAME    = Label("io.docker.compose.name")
-	PROJECT = Label("io.docker.compose.project")
-	SERVICE = Label("io.docker.compose.service")
-	HASH    = Label("io.docker.compose.config-hash")
+	NAME     = Label("io.docker.compose.name")
+	PROJECT  = Label("io.docker.compose.project")
+	SERVICE  = Label("io.docker.compose.service")
+	HASH     = Label("io.docker.compose.config-hash")
+	RECREATE = Label("io.docker.compose.recreate")
```

Comment:
>  But if docker-compose adds similar behaviour, it'll need a similar label.

It already has similar behaviour. It uses the `--no-recreate` and `--force-recreate` to handle it. With the refactor I've suggested (passing them as arguments to the recreate function, instead of through the context) this would be possible in libcompose as well.

> I'd really like to know what's wrong with this feature (marking services with desired recreate behaviour) that makes it unacceptable here

I think I've alluded to it already, but to re-state:

1. It can lead to impossible situations where you have one service that depends on another, but the service is set to never recreate. When the service is recreated, the service that is set to never recreate breaks because the links or volumes aren't pointing at the correct place anymore. How would you handle that?
2. I think the interface is wrong. It should be handled with parameters. Right now it's a magic label that compose reads but never sets. This is a deviation from all the other labels.
3. It hasn't gone through the standard review process for a new feature. Anything we add to docker-compose has a ticket created and a discussion of the design and implementation. This new feature is being mixed in with a feature parity branch, which is not the right place for it.
4.  It doesn't need to be a compose label, with parameters it could be a rancherOS label, which I think is an argument for keeping it out of compose, at least until there is some more interest in such a feature. 

> Sure we can re-implement parts of libcompose ...

I think with the refactoring I've suggested (to move it to an argument) you won't need to re-implement anything. You'll be able to store the config on the RancherOS side as a label. You can then read the label or config to determine how a conatiner should be recreated.

> is there an official document for libcompose that states it should be a subset of docker-compose, feature-wise?

The README states as the first line "A Go library for Docker Compose. It does everything the command-line tool does". It doesn't say anything about new features really.

I don't think it necessarily has to be a subset of features, but new features should go through the same design review that we have for all projects https://github.com/docker/docker/blob/master/project/REVIEWING.md#workflow